### PR TITLE
Fix borrow when array is NULL

### PR DESCRIPTION
**Pull request by benjeffery**
An assertion was triggered when NULL was passed along with KAS_BORROWS_ARRAY. This is a fix and test.

### CODE REVIEW

File: c/kastore.c [hunk 1]
```diff
@@ -908,6 +908,12 @@ kastore_bput(kastore_t *self, const char *key, size_t key_len, const void *array
     if (ret != 0) {
         goto out;
     }
+    if (array == NULL) {
```
Comment (jeromekelleher) [inline c/kastore.c:911]: Is this necessary? Isn't the problem just that the later assert is wrong (or at least should be checking that ``array_len == 0`` when both are NULL?

**Review by jeromekelleher — COMMENTED** [review]
I see - so we allow NULL as a pointer when array len is 0, but we're hitting any assert saying that the pointer must be non NULL.

Are we testiung the NULL pointer behaviour when in the non BORROW path?

**Comment by jeromekelleher** [comment]
Adding this to the milestone. I don't think we need the malloc, we just need to change the offending assertion since it's a valid thing to do.

File: c/kastore.c (hunk 1 above)
Comment (jeromekelleher) [reply c/kastore.c:911]: Changed my mind in meeting, let's keep it this way for symmetry with the other code path.

**Comment by benjeffery** [comment]
@jeromekelleher Test added to the non-borrow branch, so that they both have tests with a NULL array.

**Review by jeromekelleher — APPROVED** [review]
LGTM!

**Pull request merged**